evennia/utils/evmenu.py

```diff
@@ -1038,31 +1173,55 @@ def helptext_formatter(self, helptext):
         """
         return dedent(helptext.strip("\n"), baseline_index=0).rstrip()
 
-    def options_formatter(self, optionlist):
+    @staticmethod
+    def options_formatter(options):
         """
         Formats the option block.
 
         Args:
-            optionlist (list): List of (key, description) tuples for every
-                option related to this node.
-            caller (Object, Account or None, optional): The caller of the node.
+            options (list or dict): List of (key, description) tuples for every
+                option related to this node or dict containing the list as well
+                as various formatting instructions such as a list of options
+                to hide or move.
 
         Returns:
             options (str): The formatted option display.
 
         """
+        if isinstance(options, dict):
+            optionlist = options['display']
```

Comment:
This is not covered in the docstring. Was this meant to be `contents`?

Reply:
No. This is the ``display_options`` list generated in the ``node`` method starting at line 832 in the existing code and 988 in the proposed code. It isn't listed in the docstring because it isn't something that anyone enters (like ``display_options``)